### PR TITLE
Add debug HLD, update rest of toolchain HLD

### PR DESCRIPTION
**Pull request by craigblackmore**

### CODE REVIEW

File: docs/overlay-hld.adoc [hunk 1]
```diff
@@ -767,39 +769,44 @@ the CSV in which the grouping tool will output its groupings.
...
-* The first group contains the overlay offset table followed by the multi-group
-table (if multi-groups are present).
-* Each input section `.ovlinput.*` for each overlay symbol is copied into the
-group(s) to which it has been assigned.
-
-.ovlcache - the overlay heap into which overlay groups are loaded at runtime by
-the RT-Engine. Referred to as ``heap area'' in the requirements document.
+  * The first group contains the overlay offset table followed by the multi-group
+  table (if multi-groups are present).
+  * Each input section `.ovlinput.*` for each overlay symbol is copied into the
```
Comment (rhwd) [inline docs/overlay-hld.adoc:788]: ```  * Each input section `.ovlinput.*` for each overlay symbol is copied into the
  group(s) to which it has been assigned.```
Not sure I understand the intention

Comment (craigblackmore) [reply docs/overlay-hld.adoc:788]: Is it clearer if I change that text to `Each overlay function and data is copied to the group(s) to which it has been assigned.`?

Comment (rhwd) [reply docs/overlay-hld.adoc:788]: Yes, I think it will be clearer

Comment (craigblackmore) [reply docs/overlay-hld.adoc]: Thanks, now updated.

**Review by rhwd — APPROVED** [review]

**Pull request merged**